### DISCUSSION TITLE
Virus warning when installing using Homebrew

### DISCUSSION THREAD

**Discussion by KieranConlon** [aws/aws-sdk-cpp#3069] [Q&A] [answered]
Upgrading my homebrew packages this morning, `aws-sdk-cpp` upgrading from 1.11.360 --> 1.11.375 and `libaws-cpp-sdk-ssm-quicksetup.dylib` files was quarantined by my virus scanner before it could be installed.
Looking-up details for OSX/PSW.ext it would appear to be a password stealer trojan.

Has the community seen this before?

![image](https://github.com/user-attachments/assets/8903b797-0029-4cec-846a-184d8eeefc24)

**Comment by sbiscigl** [comment]
just tried to repro by sending the dylib to [virus total](https://www.virustotal.com/gui/file/26bb671d7867d373d1cc3201390da04237a718183a076cce482a5a4f7262e3fd/details) and it seems to be finding nothing. could you try uploading your dylib to [virus total](https://www.virustotal.com/gui/home/upload) and see if you get a different result?

**Reply by jmklix** [reply]
Also if you could let us know which virus scanner you are using?

**Reply by KieranConlon** [reply]
Upload the file to VirusTotal but it seems to stall on the computing hash stage, so not able to confirm one way or the other if it has a virus.

As for my virus scanner, I'm using Intego Virus Barrier v10.9.93 (3646) with latest virus definition file (1st Aug 2024).

Details of the VirusBarrier malware defs…

> Name: VirusBarrier Malware Definitions
Version: 08/01/24 (2024080101)
Date: Thu 1 Aug 2024, 16:31
Version 2024080101 
Latest VirusBarrier malware definitions (avlib3)
Status: The update has been completed successfully.

**Reply by KieranConlon** [reply]
Just some additional info, in case it is useful.
This install of `aws-sdk-cpp` is a dependency of another packakge.

The top-level package I installed is `gdal`, which has a dependency on `apache-arrow` which has dependency on `aws-sdk-cpp`.

**Reply by DmitriyMusatkin** [reply]
I would try again with virustotal, its one of the more comprehensive tools out there. 

In general there are 2 possibilities here:
1. VirusBarrier has a false a positive in their definitions
2. There is an active campaign to insert malicious stuff into binary distributions of cpp sdk

Given, virustotal does not flag anything in our runs on homebrew and manually build dyn lib. And also given that VirusBarrier seems to run into false positives quite often based on quick online research. I would guess that 1 is much more probable than 2. In which case it is more of a bug for VirusBarrier to address.

**Reply by KieranConlon** [reply]
Tried to analyse with Virus Total again and got a result this time - no issues found.
<img width="779" alt="image" src="https://github.com/user-attachments/assets/af3302d4-13ad-444f-8dae-ade3dcec2fd7">

So, could be a false positive with VirusBarrier.

**Reply by sbiscigl** [reply]
submitted a report with them to mark as false positive

**Reply by KieranConlon** [reply]
> submitted a report with them to mark as false positive

I did too, earlier today.

**Comment by KieranConlon** [comment, marked as answer]
I just got a reply from Intego's malware detection team and they confirm that this virus warning was a false positive.
Intego have already released an update to their malware definitions, on Aug-8th.
Re-scanning with the latest updates does not give any alert/warning.

Response from Intego…
> Our malware detection team confirmed this was not malware and added the file's necessary signature to ensure the latest malware definitions do not detect these updates, ensuring any false positive detections.
